docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+
+It is possible to generate a self-signed certificate with the ``openssl``,
+``keytool``, or on Linux, ``certtool`` commands. Such certificates are meant to
+be forwarded to a CA with a Certificate Signing Request (CSR). The CA returns a
+globally signed key or a validating certificate to add to the locally-signed
+key.
+
+Some sites have internal CAs that can provide an internal certificate for a
+self-signed key, valid only on the internal network.
+
+Self-signed certificates can be useful during development of a cluster for
+internal use only. We recommend never using a self-signed cert for a production
+Trino server, especially one that has a globally valid DNS name and address.
+
+Certificate authorities are generally willing to sign CSRs for the server name
+``localhost`` or for an internal DNS name with a restricted
+`private network <https://en.wikipedia.org/wiki/Private_network>`_ address.
+
+However, CAs do not validate CSRs for self-signed certs under globally
+accessible circumstances. For a CA to validate a CSR for a server that has a
+globally accessible DNS name and URL, the CA must be able to validate that there
+is a server at that URL, and that your company owns the specified URL and DNS
+name. This requirement prevents signed certs that are valid on one site from
+being copied and re-used on another site. Modern browsers detect cases of valid
+certs used on the wrong server, and block attempts to connect to them.
+
+You cannot use a self-signed cert without validation from a CA. Modern browsers
+detect servers running with such certs and either put up connection roadblocks
+or outright refuse to open such sites. Likewise, the Trino CLI detects a server
+running with an invalid self-signed cert and either warns against connecting or
+blocks the connection entirely.
+
+.. _tlsglossary:
+
+TLS Glossary
+------------
+
+.. _glossCA:
+
+CA
+    Certificate Authority, a trusted organization that examines and validates
+    organizations and their proposed server URIs, and issues digital
+    certificates verified as valid for the requesting organization.
+
+.. _glossCert:
+
+Certificate
+    A public key `certificate
+    <https://en.wikipedia.org/wiki/Public_key_certificate>`_ issued by a CA,
+    sometimes abbreviated as *cert*, that verifies the ownership of a server's
+    keys. Certificate format is specified in the `X.509
+    <https://en.wikipedia.org/wiki/X.509>`_ standard.
+
+.. _glossJKS:
+
+JKS
+    Java KeyStore, the system of public key cryptography supported as one part
+    of the Java security APIs. The legacy JKS system recognizes keys and
+    certificates stored in *keystore* files, typically with the ``.jks``
+    extension, and relies on a system-level list of CAs in *truststore* files
+    installed as part of the current Java installation.
+
+.. _glossKey:
+
+Key
+    A cryptographic key specified in
+    `PKCS #1 <https://en.wikipedia.org/wiki/PKCS_1>`_ as a pair of public and
```

Comment:
I would not give reference to PKCS #1 here.  That spec only covers how to encode RSA keys, but keys existed long before PKCS #1.  

I'm not sure we need this entry in the glossary.

Reply:
done